<?xml version="1.0"?>
<launch>
  <env name="ROSCONSOLE_FORMAT" value="[${node}][${time:%H:%M:%S%F}][${severity}]:${message}"/>
  <arg name="map_size_x" value="16" />
  <arg name="map_size_y" value="16" />
  <arg name="map_size_z" value="7" />
  <arg name="obs_num" value="10" />
  <arg name="num_robots" value="2" />
  <!-- Number of obstacles -->

  <group ns="uav0">
    <arg name="drone_id" value="0" />
    <arg name="pose_topic" value="/uav$(arg drone_id)/mavros/global_position/pose"/>
    <node pkg="traj_server" name="traj_server" type="bezier_traj_server" output="screen">
      <remap from="~trajectory" to="planner/trajectory" />
      <remap from="/position_cmd" to="tracker/pos_cmd" />
      <remap from="/pva_setpoint" to="tracker/pva_cmd" />
      <remap from="~odom" to="$(arg pose_topic)" />
      <param name="tf" value="true" />
      <param name="replan_time_threshold" value="1.2" />
      <param name="tf_prefix" value="uav$(arg drone_id)" />
      <param name="init_x" value="0"/>
      <param name="init_y" value="8"/>
      <param name="init_z" value="1"/>
      <param name="init_qx" value="0" />
      <param name="init_qy" value="0" />
      <param name="init_qz" value="-0.707" />
      <param name="init_qw" value="0.707" />
      <param name="is_init_yaw" value="true" />
    </node>
  </group>

  <group ns="uav1">
    <arg name="drone_id" value="1" />
    <arg name="pose_topic" value="/uav$(arg drone_id)/mavros/global_position/pose"/>
    <node pkg="traj_server" name="traj_server" type="bezier_traj_server" output="screen">
      <remap from="~trajectory" to="planner/trajectory" />
      <remap from="/position_cmd" to="tracker/pos_cmd" />
      <remap from="/pva_setpoint" to="tracker/pva_cmd" />
      <remap from="~odom" to="$(arg pose_topic)" />
      <param name="tf" value="true" />
      <param name="replan_time_threshold" value="1.2" />
      <param name="tf_prefix" value="uav$(arg drone_id)" />
      <param name="init_x" value="0"/>
      <param name="init_y" value="-8"/>
      <param name="init_z" value="1"/>
      <param name="init_qx" value="0" />
      <param name="init_qy" value="0" />
      <param name="init_qz" value="0.707" />
      <param name="init_qw" value="0.707" />
      <param name="is_init_yaw" value="true" />
    </node>
  </group>

</launch>
